test/controllers/concerns/parameters/user_test.rb

```diff
@@ -34,6 +39,27 @@ class UserParametersTest < ActiveSupport::TestCase
       params = ActionController::Parameters.new(:user => {:login => 'another'})
       assert_includes(as_admin { self.class.user_params_filter.filter_params(params, context) }, 'login')
     end
+
+    test "permits login attribute for external users with permissions" do
```

Comment:
It shouldn't allow changing the `login` attribute for Users that are not Internal, see https://github.com/theforeman/foreman/pull/4966#issuecomment-343183640 https://github.com/theforeman/foreman/pull/4966#issuecomment-343427867 , it may have security implications and it's useless unless the User changing the `login` can also change their name in LDAP.